lep/DRAFT/011_Performance_Monitor_Goals_and_Metrics.md

```diff
@@ -0,0 +1,85 @@
+#LEP 011 - Performance Monitor Goals and Metrics
+
+Date:   March 24, 2015
+
+Status: Draft
+
+Author: fffw
+
+## Abstract
+
+To continuously monitor the performance of Lantern network, we need to define dimensions and metrics clearly, and collect data to fulfill the goal at both given and get mode.
+
+We must also keep user's anonymity.
+
+## Proposal
+
+The output, is a multi-dimensional model which can pivot from any dimension. The presentation solution is TBD, hopefully there are existing services or open source projects.
+The fact to measure is data flow. For example, we can query:
+* Average time required to establish connection from Iran to Facebook last weekend
+* RTT from the given mode servers at each data center to Google
+* Total throughput from Iran to Amsterdam data center per day
+* Error rate comparison between beta version and previous release of Lantern in each country
```

Comment:
We can set an error bit to true if any error encountered during dial/read/write and report it when connection is closed.